File: src/main/java/org/jukeboxmc/inventory/DispenserInventory.java
package org.jukeboxmc.inventory;

import com.nukkitx.protocol.bedrock.data.inventory.ContainerType;
import org.jetbrains.annotations.NotNull;
import org.jukeboxmc.blockentity.BlockEntityDispenser;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class DispenserInventory extends ContainerInventory {

    public DispenserInventory( InventoryHolder holder ) {
        super( holder, -1, 9 );
    }

    @Override
    public BlockEntityDispenser getInventoryHolder() {
        return (BlockEntityDispenser) this.holder;
    }

    @Override
    public @NotNull InventoryType getType() {
        return InventoryType.DISPENSER;
    }

    @Override
    public @NotNull ContainerType getWindowTypeId() {
        return ContainerType.DISPENSER;
    }
}

File: src/main/java/org/jukeboxmc/inventory/DispenserInventory.kt
package org.jukeboxmc.inventory

import com.nukkitx.protocol.bedrock.data.inventory.ContainerType
import org.jukeboxmc.blockentity.BlockEntityDispenser

/**
 * @author LucGamesYT
 * @version 1.0
 */
class DispenserInventory(holder: InventoryHolder?) : ContainerInventory(holder, -1, 9) {
    override val inventoryHolder: InventoryHolder?
        get() = holder as BlockEntityDispenser
    override val type: InventoryType
        get() = InventoryType.DISPENSER
    override val windowTypeId: ContainerType
        get() = ContainerType.DISPENSER
}